Provide the content for the Diazo theming part

erral opened the pull request:
Can be seen as part of #1645

<!-- readthedocs-preview plone6 start -->
----
📚 Documentation preview 📚: https://plone6--1870.org.readthedocs.build/

<!-- readthedocs-preview plone6 end -->

stevepiercy committed:
Update tips submodules/plone.api submodules/volto

stevepiercy committed:
Update tip submodules/volto

stevepiercy committed:
Fix broken references (#1868)

* Fix broken references
- `link.svg`
- `_inc/_install-browser-reqs-volto.md`

* Fix linkcheck to bobtemplates

* Update tip submodules/volto

* Fix link

stevepiercy committed:
Update tip submodules/plone.api

stevepiercy committed:
Update tip submodules/volto

erral committed:
initial structure of Theming with Diazo

erral committed:
initial structure of Theming with Diazo

erral committed:
more doc

erral committed:
more doc

stevepiercy committed:
Merge remote-tracking branch 'origin/erral-diazo-docs' into erral-diazo-docs

stevepiercy committed:
- Add diazo to toctree
- Add note update getting help for choosing a theming method

stevepiercy committed:
Clean up grammar, one sentence per line, MyST syntax

stevepiercy commented on docs/classic-ui/theming/diazo.md:
```diff
@@ -0,0 +1,136 @@
+
+## Create an add-on package
+
+To create a Diazo theme, you need to create an add-on package with [`plonecli`](https://github.com/plone/plonecli).
```
What are the pre-requisites to running `plonecli`?

erral replied:
I have reworded this stating that you don't need to install plonecli, and you can run it using pipx, like we run cookieplone: (pipx run plonecli)

stevepiercy replied:
Sweet! Old way meets new!

stevepiercy committed:
Add language

stevepiercy commented on docs/classic-ui/theming/diazo.md:
```diff
@@ -0,0 +1,136 @@
+## How to theme using Diazo
+
+You can start with the provided {file}`rules.xml` file.
+You will need to write your own rules to bring the dynamic content from Plone into the theme.
```
How will I write any rules, when there is no reference documentation linked here?

erral replied:
The reference of writing those rules is at https://docs.diazo.org/en/latest/basic.html

erral replied:
I have added a link to Diazo docs.

stevepiercy commented on docs/classic-ui/theming/diazo.md:
```diff
@@ -0,0 +1,136 @@
+For instance, you may need to put together the main menu, the language change, and the search box.
+Sometimes it is easier to override the corresponding template in Plone, build the new HTML structure there, and replace one thing in the {file}`rules.xml` file than trying to write complex Diazo rules or writing XSLT.
+
+The size of the {file}`rules.xml` file and the number of rules it contains can negatively impact the performance of your site.
```
That won't be a problem, since there is no reference to how to write these rules. 🙃

erral replied:
See comment above :wink:

stevepiercy commented on docs/classic-ui/theming/diazo.md:
```diff
@@ -0,0 +1,136 @@
+└── tinymce-templates
+    ├── bs-dark-hero.html
+    ├── bs-hero-left.html
+    └── bs-pricing.html
```
It would be good to describe the purpose of each of these files here.

stevepiercy committed:
Move concepts and advice into a new Theme development advice section

stevepiercy committed:
Minor tweaks

stevepiercy committed:
Rename section to reflect its content

stevepiercy committed:
Adjust heading level

stevepiercy reviewed (changes requested):
I restructured the Diazo doc to present the critical path first, followed by advice. It flows a little better, but there are still some issues with it, namely a first-timer would have no clue what to do after reading this.

erral committed:
Update diazo.md

explain the purpose of each file created when addin a theme

erral committed:
Update diazo.md

Link to Diazo docs.

erral committed:
Update diazo.md

stevepiercy committed:
- Make first mention of Diazo a term, already existing in the Glossary
- Reuse a sentence from Diazo docs that fills a critical gap in the introduction.
- Convert list of files descriptions into a definition list

stevepiercy committed:
Tweak markup

stevepiercy commented on docs/classic-ui/theming/diazo.md:
```diff
@@ -0,0 +1,178 @@
+
+{file}`tinymce-templates`
+:   Template files that can be loaded into the TinyMCE editor in Plone.
+    It requires additional configuration in the add-on's profile's {file}`registry.xml` file.
```
I searched within the Classic UI folder for a reference to the file `registry.xml`, but came up empty. Is that the correct filename? There is documentation on TinyMCE in https://6.docs.plone.org/classic-ui/tinymce-customization.html that uses the filename `profiles/default/registry/tinymce.xml`.

Where can I get more information? We should add a `seealso` admonition to whatever that reference should be, in case someone wants to fiddle with TinyMCE. Here's my guess. Please let me know. Thank you!

````suggestion
    It requires additional configuration in the add-on's profile's registry.

    ```{seealso}
    {ref}`classic-ui-tinymce-customization-insert-preconfigured-html-blocks-label`
    ```
````

erral replied:
The file `registry.xml` present in `profiles/default` can be split in several files under `profiles/default/registry` with arbitrary names ending in `.xml`, so `registry.xml` and `profiles/default/registry/tinymce.xml` is valid and `profiles/default/registry/plone.xml` is valid.

The docs lack information about tinymce templates. I will add a PR with that information and then we can link there.

erral replied:
@stevepiercy I have added the relevant information in the tinymce docs in this very PR. 

I tried to use the same example content as created by the plonecli template. 

Let's see if it is understandable :)

stevepiercy replied:
Yup, this is great! I tidied it up a bit. However, you should take a look at a compatibility issue with Plone 6.1/TinyMCE 7 and the Template plugin in https://plone6--1870.org.readthedocs.build/classic-ui/tinymce-customization.html#insert-preconfigured-html-blocks. 😞 

I also revised the suggestion, and added your comment as a tip to the TinyMCE docs.

Please let me know. Thank you!

erral replied:
Then we better add a warning saying that this only works for Plone 6.0 and not 6.1 right?

stevepiercy replied:
Good question. There was a [similar issue with the accordion plugin](https://6.docs.plone.org/backend/upgrading/version-specific-migration/upgrade-to-61.html#tinymce-upgraded-in-classic-ui). It appears that there was work on the Template plugin in mockup in https://github.com/plone/mockup/pull/1384. @petschki can you advise?

I don't know if this requires an entry in the 6.1 upgrade guide.

petschki replied:
the `template` plugin has moved to a premium plugin since tiny version `6.4` (see https://www.tiny.cloud/docs/tinymce/latest/advanced-templates/) but we've reimplemented the "community template plugin" for version 7.x in mockup. So its there and it works in Plone 6.1 too.

stevepiercy reviewed (changes requested):
Almost there! A couple of questions regarding the `registry.xml` file for TinyMCE.

stevepiercy committed:
Merge branch 'classic-ui-theming-reorg-1645' into erral-diazo-docs

stevepiercy committed:
Merge branch 'classic-ui-theming-reorg-1645' into erral-diazo-docs

# Conflicts:
#	docs/classic-ui/theming/index.md

erral committed:
add tinymce templates docs

stevepiercy committed:
Tidy up

stevepiercy committed:
Add a label and important admonition about Template plugin removal in TinyMCE 7.0

petschki commented on docs/classic-ui/tinymce-customization.md:
````diff
@@ -168,3 +167,46 @@ You can also exclude certain URLs from being sandboxed as follows.
+You can add custom HTML blocks in TinyMCE.
+You can insert them in your content using the TinyMCE {menuselection}`Insert Template` menu option using the [TinyMCE Template plugin](https://www.tiny.cloud/docs/tinymce/6/template/).
+
+```{important}
+The Template plugin has been completely removed from the TinyMCE 7.0 release used in Plone 6.1.
+As an alternative solution, you can use the [Advanced Template Premium plugin](https://www.tiny.cloud/docs/tinymce/6/advanced-templates/). 
+```
+
````
This can be removed since we've backported the community edition of `template` plugin to TinyMCE 7

stevepiercy replied:
Done in https://github.com/plone/documentation/pull/1870/commits/2d47570f0de137adfdaee38dbfed18bd47ed1888. Thanks @petschki! Will merge the chain of PRs when all green.

stevepiercy committed:
Remove important admonition about Template plugin

stevepiercy commented:
The failing test is a linkcheck failure that is fixed in the related PR. Merging! 🎉 Thanks @erral for all the work, and @petschki for the assist!
